tutorials/18_training_interface.ipynb

```diff
@@ -303,63 +284,193 @@
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "5a5b9e60-f504-4837-9ce8-c5af63dd03bd",
+   "id": "d3c92a48-d6ff-4e06-a842-47c0e07de217",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "### Custom Data Loaders\n",
+    "\n",
+    "One helpful advantage of having access to the training loop is, that you can now use your own DataLoaders during training of the density estimator. In this fashion, larger datasets can be used as input to `sbi` where `x` is potentially an image or something else. While this will require [embedding the input data](https://sbi-dev.github.io/sbi/latest/tutorials/04_embedding_networks/), a more fine grained control over loading the data is possible and allows to manage the memory requirement during training.\n",
+    "\n",
+    "First, we build a Dataset that complies with the `torch.util.data.Dataset` API. Note, the class below is meant for illustration purposes. In practice, this class can also read the data from disk etc.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8dbe6e31-e682-419f-b88f-23b1e833eaf8",
    "metadata": {},
+   "outputs": [],
+   "source": [
+    "class NPEData(torch.utils.data.Dataset):\n",
+    "\n",
+    "    def __init__(self, nsamples:int,\n",
+    "                 prior: torch.distributions.Distribution ,\n",
+    "                 simulator: Callable,\n",
+    "                 seed:int = 44):\n",
+    "        super().__init__()\n",
+    "\n",
+    "        torch.random.manual_seed(seed) #will set the seed device wide\n",
+    "        self.prior = prior\n",
+    "        self.simulator = simulator\n",
+    "\n",
+    "        self.theta = prior.sample((nsamples,))\n",
+    "        self.x = simulator(self.theta)\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return self.theta.shape[0]\n",
+    "\n",
+    "    def __getitem__(self, index:int):\n",
+    "        return self.theta[index,...], self.x[index,...]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d8178235-291c-43e7-8376-98e42542d42d",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "We can now proceed to create a DataLoader and conduct our training loop as illustrated above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eb2aa734-dfca-4691-b3b5-f5be889219ba",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data = NPEData(2048, prior, simulator)\n",
+    "train_loader = torch.utils.data.DataLoader(train_data, batch_size=128)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2759a702-6ce7-4e2c-8373-402fb30d7f49",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "For sake of demonstration, let's create another estimator using a masked autoregressive flow (maf). For this, we create a second dataset and use only parts of the data to construct the maf estimator."
```

Comment:
```suggestion
    "For sake of demonstration, let's create another estimator using a masked autoregressive flow (maf). For this, we create a second dataset and use only parts of the data to construct the `maf` estimator (e.g., to avoid memory overflow for large datasets)."
```